File: src/main/java/kitchenpos/eatinorders/tobe/domain/EatInOrderLineItem.java
package kitchenpos.eatinorders.tobe.domain;

import jakarta.persistence.Column;
import jakarta.persistence.Entity;
import jakarta.persistence.GeneratedValue;
import jakarta.persistence.GenerationType;
import jakarta.persistence.Id;
import jakarta.persistence.Table;
import kitchenpos.menus.tobe.domain.MenuPrice;

import java.util.UUID;

@Table(name = "order_line_item")
@Entity
public class EatInOrderLineItem {

    @Id
    @GeneratedValue(strategy = GenerationType.IDENTITY)
    @Column(name = "seq")
    private Long seq;

    @Column(name = "menu_id", nullable = false, columnDefinition = "binary(16)")
    private UUID menuId;

    @Column(name = "price", nullable = false)
    private MenuPrice menuPrice;

    @Column(name = "quantity", nullable = false)
    private long quantity;

    protected EatInOrderLineItem() {
    }

    private EatInOrderLineItem(UUID menuId, MenuPrice menuPrice, long quantity) {
        this.menuId = menuId;
        this.menuPrice = menuPrice;
        this.quantity = quantity;
    }

    public static EatInOrderLineItem create(UUID menuId, MenuPrice menuPrice, long quantity) {
        if (quantity <= 0) {
            throw new IllegalArgumentException("주문 수량은 0보다 커야 합니다.");
        }
        return new EatInOrderLineItem(menuId, menuPrice, quantity);
    }
}

File: src/main/java/kitchenpos/eatinorders/tobe/domain/EatInOrderLineItem.kt
package kitchenpos.eatinorders.tobe.domain

import jakarta.persistence.*
import kitchenpos.common.Price
import java.util.*

@Table(name = "eat_in_order_line_items")
@Entity
class EatInOrderLineItem(
    menuId: UUID,
    quantity: EatInOrderLineItemQuantity,
    price: Price
) {
    @Column(name = "seq")
    @GeneratedValue(strategy = GenerationType.IDENTITY)
    @Id
    val seq: Long = 0L

    @Column(name = "menu_id", columnDefinition = "binary(16)", nullable = false)
    val menuId: UUID = menuId

    @Column(name = "quantity", nullable = false)
    val quantity: EatInOrderLineItemQuantity = quantity

    @Column(name = "price", nullable = false)
    val price: Price = price

    @ManyToOne
    @JoinColumn(name = "eat_in_order_id", nullable = false, columnDefinition = "binary(16)")
    var eatInOrder: EatInOrder? = null

    override fun equals(other: Any?): Boolean {
        if (this === other) return true
        if (javaClass != other?.javaClass) return false

        other as EatInOrderLineItem

        return seq == other.seq
    }

    override fun hashCode(): Int {
        return seq.hashCode()
    }

}
